src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -404,4 +404,34 @@ public function validateUrlAction(int $id, Request $request, UrlValidation $urlV
 
         return $urlValidation->validateUrl($erddapUrl);
     }
+
+    /**
+     * Returns a list of global ingest folders.
+     *
+     * @param Request           $request           The request object.
+     *
+     * @Route(
+     *     "/api/dataset_submission_folder_list",
+     *     name="pelagos_api_get_folder_list_dataset_submission",
+     *     methods={"GET"},
+     *     defaults={"_format"="json"},
+     *     )
+     *
+     * @View()
+     *
+     * @return Response
+     */
+    public function getGlobalIngestFolders(Request $request): Response
+    {
+        $path = $request->get('user');
```

Comment:
User name is not need for this request. We can get it from Symfony using: `$this->security->getUser();`
I don't think there would be a need to get this for another user by someone else.
So, then you can also get rid of the cookie reading function.

Technically we could do this in the Form, and prepopulate it there, but I kinda like the Javascrip approach, in case they are uploading on another screen or so. In that case, I would make sure the list is fetch on every time. Like on an `onChange` event.

Reply:
Made the required changes